taskt/Core/Automation/Commands/GetRegexMatchesCommand.cs

```diff
@@ -51,19 +51,23 @@ public override void RunCommand(object sender)
             var engine = (Core.Automation.Engine.AutomationEngineInstance)sender;
             var vInputData = v_InputData.ConvertToUserVariable(engine);
             string vRegex = v_Regex.ConvertToUserVariable(engine);
-            Match[] matches = Regex.Matches(vInputData, vRegex)
-                       .Cast<Match>()
-                       .ToArray();
-            var arr = string.Join(",", matches.AsEnumerable());
+
+            var matches = (from match in Regex.Matches(vInputData, vRegex).Cast<Match>() select match.Groups[0].Value).ToList();
 
             Script.ScriptVariable outputMatches = new Script.ScriptVariable
             {
                 VariableName = v_OutputVariable,
-                VariableValue = arr
+                VariableValue = matches
             };
 
-            engine.VariableList.Add(outputMatches);
+            //Overwrites variable if it already exists
+            if (engine.VariableList.Exists(x => x.VariableName == outputMatches.VariableName))
+            {
+                Script.ScriptVariable temp = engine.VariableList.Where(x => x.VariableName == outputMatches.VariableName).FirstOrDefault();
+                engine.VariableList.Remove(temp);
+            }
 
+            engine.VariableList.Add(outputMatches);
```

Comment:
This is another place where we should be using `engine.AddVariable()` for the sake of consistency.

Reply:
The changes have been accommodated.